mava/components/jax/updating/checkpointer.py

```diff
@@ -0,0 +1,95 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Callable, Optional
+
+from acme.jax import savers as acme_savers
+from chex import dataclass
+
+from mava.components.jax import Component
+from mava.core_jax import SystemParameterServer
+from mava.wrappers import SaveableWrapper
+
+"""Checkpointer component for Mava systems."""
+
+
+@dataclass
+class CheckpointerConfig:
+    checkpoint_minute_interval: float = 5
+
+
+class Checkpointer(Component):
+    def __init__(
+        self,
+        config: CheckpointerConfig = CheckpointerConfig(),
+    ):
+        """Component for checkpointing system variables."""
+        self.config = config
+
+    def on_parameter_server_init(self, server: SystemParameterServer) -> None:
+        """Create the system checkpointer.
+
+        Args:
+            server: SystemParameterServer.
+
+        Returns:
+            None.
+        """
+        saveable_parameters = SaveableWrapper(server.store.parameters)
+        server.store.system_checkpointer = acme_savers.Checkpointer(
+            object_to_save=saveable_parameters,  # must be type saveable
+            directory=server.store.experiment_path,
+            add_uid=False,
+            time_delta_minutes=0,
+        )
+        # Assign the restored parameters to parameter server
+        for key in saveable_parameters.state.keys():
```

Comment:
Can we remove this assignment by changing the `SaveableWrapper` as shown in my other comment?

Reply:
Good idea! Done

Reply:
Thanks :slightly_smiling_face: